### PR TITLE
unix: Support Android libuv standalone build

### PR DESCRIPTION
**Pull request by linusmartensson**
Tested most of my compilation in the previous patch on NodeJS
and extracted the patches from there. This additional patch ensures 
libuv will be capable of building standalone as well, both with gyp
and Makefiles.

Build documentation was added to the README.md file.

Some tests are failing, and I have not heavily investigated
the reasons. The failures are generally on error-conditions, and are
likely related to differences between fully POSIX-compatible
systems and android. Resolving these failures should logically
be done in separate bug reports and is currently out of scope
for my work on uv and node.

### CODE REVIEW

File: include/uv-private/uv-unix.h [hunk 1]
```diff
@@ -37,8 +37,17 @@
...
-#ifdef ANDROID
+#ifdef __ANDROID__
 #include "pthread-fixes.h"
+
+/*More missing defines on android*/
+#ifndef S_IREAD
+#define S_IREAD 0
+#endif
+#ifndef S_IWRITE
+#define S_IWRITE 0
```
Comment (bnoordhuis) [inline include/uv-private/uv-unix.h]: I assume that's to patch up the test suite? That should be addressed in a different way; it doesn't belong in a public header and besides, S_READ and S_IWRITE are obsolete BSD-isms anyway. I think the reason we use them is that Windows doesn't have S_IRUSR and S_IWUSR.

See #821 - once it checks out on Windows, I'll land it in master. Can you rebase and remove this afterwards?

File: config-unix.mk [hunk 2]
```diff
@@ -104,6 +104,18 @@ OBJS += src/unix/linux-core.o \
         src/unix/proctitle.o
 endif
 
+ifeq (android,$(PLATFORM))
+CFLAGS += -D_GNU_SOURCE -std=c99
```
Comment (bnoordhuis) [inline config-unix.mk]: Why -std=c99?

File: include/uv-private/uv-unix.h (hunk 1 above)
Comment (linusmartensson) [inline include/uv-private/uv-unix.h]: Will do

File: config-unix.mk (hunk 2 above)
Comment (linusmartensson) [inline config-unix.mk]: The android toolchain utilizes inline, which is not available in c89. 
c99 was my knee-jerk response to that. I checked now, and I can also leave this line out completely for a working gnu90 compilation. I'll do that instead.

**Comment by bnoordhuis** [comment]
Landed in 3fdd2a1, thanks!

**Pull request closed without merging**